File: Python/Binary-Search.py
def binary_search(array, target):
    left, right = 0, len(array) - 1

    while left <= right:
        mid = left + (right - left) // 2
        if array[mid] == target:
            return mid
        elif array[mid] < target:
            left = mid + 1
        else:
            right = mid - 1

    return -1  # Indicates that the target was not found

array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
target = 7
result = binary_search(array, target)

if result != -1:
    print("Element found at index:", result)
else:
    print("Element not found")

File: Java/BinarySearch.java
//Binary Search Algo:

public class BinarySearch {
    public static void main(String[] args) {
        int [] arr = {-212, -123, -102, -56,0,12, 29, 64, 78, 89};
        int target = 13;
        int ans = binarySearch(arr,target);
        System.out.println(ans);
    }
    static int binarySearch(int [] arr, int target) {
        int start = 0;
        int end = arr.length -1;
        while (start<=end) {
            int mid = start + (end-start)/2;

            if (target < arr[mid]) {
                end = mid-1;
            } else if (target > arr[mid]) {
                start = mid+1;
            } else {
                return mid;
            }
        }
        return -1;
    }
}
